streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -674,6 +676,8 @@ class StreamingContext private[streaming] (
           logWarning("StreamingContext has already been stopped")
         case ACTIVE =>
           scheduler.stop(stopGracefully)
+          // De-registering Streaming Metrics of the StreamingContext
+          env.metricsSystem.removeSource(streamingSource)
```

Comment:
What I mean is to move this line out of `ACTIVE` condition, In any case have to deregister the metrics source, not only in `ACTIVE` state.